debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCArrayletObjectModelBase.java

```diff
@@ -315,6 +350,68 @@ public boolean isInlineContiguousArraylet(J9IndexableObjectPointer array) throws
 		return getArrayLayout(array) == GC_ArrayletObjectModelBase$ArrayLayout.InlineContiguous;
 	}
 
+	/**
+	 * Determine if the arraylet data of arrayPtr is adjacent to its header.
+	 *
+	 * @param arrayPtr array object
+	 * @throws CorruptDataException If there's a problem accessing the indexable object dataAddr field
+	 * @return true if the arrayPtr has its data adjacent to its header
+	 */
+	public boolean isArrayletDataAdjacentToHeader(J9IndexableObjectPointer array) throws CorruptDataException
+	{
+		UDATA dataSizeInBytes = getDataSizeInBytes(array);
+		return isArrayletDataAdjacentToHeader(dataSizeInBytes);
+	}
+
+	/**
+	 * Determine if the arraylet data of arrayPtr is adjacent to its header.
+	 *
+	 * @param arrayPtr array object
+	 * @throws CorruptDataException if there's a problem accessing the indexable object dataAddr field
+	 * @return true if the arrayPtr has its data adjacent to its header
+	 */
+	public boolean isArrayletDataAdjacentToHeader(UDATA dataSizeInBytes)  throws CorruptDataException
+	{
+		MM_GCExtensionsPointer extensions = GCBase.getExtensions();
+		UDATA minimumSpineSizeAfterGrowing = new UDATA(ObjectModel.getObjectAlignmentInBytes());
+
+		return largestDesirableArraySpineSize.eq(UDATA.MAX) ||
+        dataSizeInBytes.lte(largestDesirableArraySpineSize
+							.sub(minimumSpineSizeAfterGrowing)
+							.sub(J9IndexableObjectHelper.contiguousHeaderSize()));
+	}
+
+	/**
+	 * Check if the given indexable object is double mapped.
+	 *
+	 * @param address indexable object to check if it is double mapped
+	 * @throws CorruptDataException if there's a problem accessing card table heap fields
+	 * @return true if the given indexable object is double mapped
+	 */
+	public boolean isIndexableObjectDoubleMapped(VoidPointer indexableDataAddr, UDATA dataSizeInBytes) throws CorruptDataException
```

Comment:
like in runtime variant of this method is strangly named. it's too specific name, while it has a more generic meaning (data portion of the object in outside of heap)